synapse/replication/http/keys.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2023 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import TYPE_CHECKING, Dict, List, Tuple
+
+import attr
+from signedjson.key import decode_verify_key_bytes, encode_verify_key_base64
+from unpaddedbase64 import decode_base64
+
+from twisted.web.server import Request
+
+from synapse.crypto.types import _FetchKeyRequest
+from synapse.http.server import HttpServer
+from synapse.replication.http._base import ReplicationEndpoint
+from synapse.storage.keys import FetchKeyResult
+from synapse.types import JsonDict
+from synapse.util.async_helpers import yieldable_gather_results
+
+if TYPE_CHECKING:
+    from synapse.server import HomeServer
+
+logger = logging.getLogger(__file__)
+
+
+class ReplicationFetchKeysEndpoint(ReplicationEndpoint):
+    """Another worker is asking us to fetch keys for a homeserver X.
+
+    The request looks like:
+
+        POST /_synapse/replication/fetch_keys
+        {
+            keys_to_fetch: [
+                {
+                    "server_name": "example.com",
+                    "minimum_valid_until_ts": 123456,
+                    "key_ids": ["ABC", "DEF"]
+                }
+            ]
+        }
+
+    We would normally return a group of FetchKeyResponse structs like the
+    normal code path does, but FetchKeyResponse holds a nacl.signing.VerifyKey
+    which is not JSON-serialisable. Instead, for each requested key we respond
+    with a boolean: `true` meaning we fetched this key, and `false` meaning we
+    didn't.
+
+    The response takes the form:
+
+        200 OK
+        {
+            "fetched_count": 1
+        }
```

Comment:
Are these comments still accurate? I can't figure out what the endpoint returns just from the code below, but it doesn't seem to match the comment.

Reply:
No, I think I had iterated since then and not updated the comment.